### PR TITLE
feat(ui): bottom nav

### LINKED ISSUE #202: Option to move navigation at the bottom of the screen (on mobile)
I use Pinafore on mobile a lot, and I would love to have the navigation bar at the bottom of the screen for easier reach.

I suggest adding a new setting to enable that. What do you think?

### PR DESCRIPTION
**Pull request by bennypowers**
Closes #202 

I had some difficulty testing locally

```
$ yarn run-mastodon
yarn run v1.22.19
$ node ./bin/run-mastodon.js
Installing Mastodon...
Setting up mastodon database...
ChildProcessError: Command failed: createdb -h 127.0.0.1 -U pinafore -w pinafore_development
createdb: error: connection to server at "127.0.0.1", port 5432 failed: FATAL:  Ident authentication failed for user "pinafore"
 `createdb -h 127.0.0.1 -U pinafore -w pinafore_development` (exited with error code 1)
    at callback (/home/bennyp/Developer/pinafore/node_modules/child-process-promise/lib/index.js:33:27)
    at ChildProcess.exithandler (node:child_process:406:5)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 1,
  childProcess: {
    _forkChild: [Function: _forkChild],
    ChildProcess: [Function: ChildProcess],
    exec: [Function: exec],
    execFile: [Function: execFile],
    execFileSync: [Function: execFileSync],
    execSync: [Function: execSync],
    fork: [Function: fork],
    spawn: [Function: spawn],
    spawnSync: [Function: spawnSync]
  },
  stdout: '',
  stderr: 'createdb: error: connection to server at "127.0.0.1", port 5432 failed: FATAL:  Ident authentication failed for user "pinafore"\n'
}
error Command failed with exit code 1.
```

that was after installing enabling and starting postgres, then `CREATE USER mastodon CREATEDB;` and `CREATE USER pinafore CREATEDB;`

:shrug:

### CODE REVIEW

**Comment by nolanlawson** [comment]
@bennypowers Please see the `create user` query here: https://github.com/nolanlawson/pinafore/blob/master/CONTRIBUTING.md#running-integration-tests

Last I checked, I also had to run this in an Ubuntu 18.04 container in Podman to get this to work. I should probably try to improve how these tests are run.

Thanks for the PR! I was out of town the past two weeks but I will take a look soon.

File: src/routes/_store/observers/bottomNavObservers.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export function bottomNavObservers (store) {
+  store.observe('bottomNav', bottomNav => {
+    // disables or enables the style
+    document.body.classList.toggle('bottom-nav', bottomNav);
+    document.querySelector('.main-content').classList.toggle('bottom-nav', bottomNav);
+    document.getElementById('main-nav').classList.toggle('bottom-nav', bottomNav);
```
Comment (nolanlawson) [inline src/routes/_store/observers/bottomNavObservers.js:6]: It's not clear to me why we need to set this in three different places.

Historically, the way I've done this is to toggle the entire `<style>` tag itself. You can see some examples here:

https://github.com/nolanlawson/pinafore/blob/b152359428c9582c0a212f0872cc805cfb8c08ad/src/build/template.html#L35-L44

https://github.com/nolanlawson/pinafore/blob/16e66346d7582ffddf27112706f834a4a0fbbbcc/src/routes/_store/observers/grayscaleObservers.js#L5-L16

File: src/scss/global.scss [hunk 2]
```diff
@@ -20,7 +20,7 @@ body {
 
 .main-content {
   contain: content; // see https://www.w3.org/TR/2018/CR-css-contain-1-20181108/#valdef-contain-content
-  padding-top: var(--main-content-pad-top);
+  padding-block-start: var(--main-content-pad-top);
```
Comment (nolanlawson) [inline src/scss/global.scss:23]: Is this orthogonal to the PR? Does this have an effect with different languages? (I can't think of any bottom-to-top languages, but I'm sure they exist?)

File: src/routes/_store/observers/bottomNavObservers.js (hunk 1 above)
Comment (bennypowers) [reply src/routes/_store/observers/bottomNavObservers.js:6]: agreed. i'll have to pull up my sleeves and try to get my local env running so i can clean this up

File: src/scss/global.scss (hunk 2 above)
Comment (bennypowers) [reply src/scss/global.scss:23]: you could take it out :shrug:

**Comment by nolanlawson** [comment]
Closing in favor of #2205, thanks for opening this PR! :bow:

**Pull request closed without merging**